src/lib/support/CHIPSparseArray.h

```diff
@@ -0,0 +1,193 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    Copyright (c) 2016-2017 Nest Labs, Inc.
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <stddef.h>
+
+#include <core/CHIPEncoding.h>
+#include <core/CHIPError.h>
+#include <support/Base64.h>
+#include <support/CHIPMem.h>
+#include <support/CodeUtils.h>
+#include <support/SafeInt.h>
+#include <transport/raw/MessageHeader.h>
+
+namespace chip {
+
+template <uint16_t Capacity>
```

Comment:
nit: since this is a constant value, naming kCapacity may be best.

Some doccomment about use would be nice:
   - what is a 'sparse array'? This looks more like a 'set' to me, so how about 'SerializableUint64Set'? Do we only need uint64?
   - what were the design choices/intent?